### PR TITLE
AppConfig samples

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Create, Retrieve and Delete a Secret Reference
+
+The AppConfiguration service supports KeyVault Secret reference settings.
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:3]: From an outsider, I guess it is not very clear what is the advantage of using this. Maybe make some emphasis on the way you can connect both services and retrieve/manage keys safely? or min interaction/modification?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Secret references are settings that follow specific JSON schema for the `Value`, and the `application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8` content type.
+The `Azure.Data.AppConfiguration` library provides a strongly-typed way of managing Secret references.
+This sample shows how to use the library to create, retrieve, and delete Secret references.
+
+## Create a Secret Reference
+
+To create a secret reference, use the `SecretReferenceConfigurationSetting` class:
+
+```C# Snippet:Sample_CreateSecretReference
+var secretReferenceSetting = new SecretReferenceConfigurationSetting("setting", new Uri("https://<keyvault_name>.vault.azure.net/secrets/<secret_name>"));
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:13]: is the Uri a value the user can get in any way? 
I worry if the user wants to use another cloud, that they might think that the format of the Uri is always this and will miss that the DNS suffix could vary

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Create, Retrieve and Delete a Secret Reference
+
+The AppConfiguration service supports KeyVault Secret reference settings.
+Secret references are settings that follow specific JSON schema for the `Value`, and the `application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8` content type.
+The `Azure.Data.AppConfiguration` library provides a strongly-typed way of managing Secret references.
+This sample shows how to use the library to create, retrieve, and delete Secret references.
+
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:7]: Samples usually start with a [Create client](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample1_HelloWorld.md#create-a-configurationclient) section

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```C# Snippet:Sample_SetSecretReference
+client.SetConfigurationSetting(secretReferenceSetting);
+```
+
+## Retrieve a Secret Reference
+
+You can use the `GetConfigurationSetting` method to retrieve the secret reference.
+
+If you'd like to get the secret value you would need to reference the [Azure.Security.KeyVault.Secrets](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/README.md) library and use the `KeyVaultSecretIdentifier` to parse the `SecretId`, and `SecretClient.GetSecretAsync` to get the secret value.
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:30]: Do you think a user will care which version of KV.Secrets this sample uses? I like it this way so we don't have to manually maintain a reference here. At the same time wonder if a user will care and will be able to find the version reference in our repo.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample8_ChangeNotification.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# Create, Retrieve and Delete a Configuration Setting
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample8_ChangeNotification.md:1]: Looks like tittle was not updated

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/samples/Sample10_SecretReference.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using Azure.Core.TestFramework;
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+
+namespace Azure.Data.AppConfiguration.Samples
+{
+    public partial class ConfigurationSamples: SamplesBase<AppConfigurationTestEnvironment>
+    {
+        public async Task CreateSecretReference()
```
Comment (maririos) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/samples/Sample10_SecretReference.cs:15]: is it intentional that this is not a running test?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 2 above)
Comment (pakrym) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:13]: Yeah, both via the portal and the KV SDK.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 4 above)
Comment (pakrym) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:30]: I'll add a comment about when the  `KeyVaultSecretIdentifier` feature was added.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 3 above)
Comment (pakrym) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:14]: Do you think it's worth duplicating the same section everywhere?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 1 above)
Comment (pakrym) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md]: Added a few lines from docs.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 3 above)
Comment (maririos) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:14]: The idea to have it in each file is that the user won't need to jump across files to have basic behavior.
I guess to your point, this new samples are advanced samples so might not be needed.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md (hunk 2 above)
Comment (maririos) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md:20]: Consider adding a comment so users can know where to get that value from

**Review by maririos — APPROVED** [review]

**Pull request merged**